Lib/logging/__init__.py

```diff
@@ -1543,9 +1556,15 @@ def isEnabledFor(self, level):
         """
         Is this logger enabled for level 'level'?
         """
-        if self.manager.disable >= level:
-            return False
-        return level >= self.getEffectiveLevel()
+        if level not in self._cache:
+            _acquireLock()
+            if self.manager.disable >= level:
+                self._cache[level] = False
+            else:
+                self._cache[level] = level >= self.getEffectiveLevel()
+            _releaseLock()
+
+        return self._cache[level]
```

Comment:
The real race condition is if clear_cache() is called in-between. But, yes, your proposed idiom would fix it.